### PR TITLE
Add position error into deformable vs. rigid contact

### PR DESCRIPTION
**Pull request by xhan0619**
Add position error into deformable vs. rigid contact where the error is added directly to the constraint (i.e. not splitImpulse). 

Change the default damping model for mass pring to the angular momentum one for more accuracy and visual correctness.

Fix a bug in the interpolation transform of the rigid body collision wrapper.

### CODE REVIEW

**Comment by erwincoumans** [comment]
Thanks a lot for all the improvements. Let me review the changes and get back to you.

File: src/BulletSoftBody/btDeformableContactConstraint.cpp [hunk 1]
```diff
@@ -256,6 +256,8 @@ btScalar btDeformableRigidContactConstraint::solveConstraint()
     impulse = impulse_normal + impulse_tangent;
     // apply impulse to deformable nodes involved and change their velocities
     applyImpulse(impulse);
+	if (residualSquare < 1e-7)
```
Comment (erwincoumans) [inline src/BulletSoftBody/btDeformableContactConstraint.cpp]: Could you describe the idea behind this early out?

File: src/BulletSoftBody/btSoftBody.cpp [hunk 2]
```diff
@@ -3681,8 +3698,8 @@ void btSoftBody::defaultCollisionHandler(const btCollisionObjectWrapper* pcoWrap
                     docollideFace.psb = this;
                     docollideFace.m_colObj1Wrap = pcoWrap;
                     docollideFace.m_rigidBody = prb1;
-                    docollideFace.dynmargin = basemargin + timemargin;
-                    docollideFace.stamargin = basemargin;
+					docollideFace.dynmargin = 0.05*(basemargin + timemargin);
```
Comment (erwincoumans) [inline src/BulletSoftBody/btSoftBody.cpp]: Since it is still in research stage, I'll LGTM, , Do we want to expose those constants?

**Comment by erwincoumans** [comment]
Thanks for the contribution!
I'm fine with merging the code. Some of the features are conditionally compiled, we may want to enable/disable them at run-time? 
(#ifdef USE_STRAIN_RATE_LIMITING and #define USE_QUADRATURE 1, //#define CACHE_PREV_COLLISION etc)

File: src/BulletSoftBody/btDeformableContactConstraint.cpp (hunk 1 above)
Comment (xhan0619) [reply src/BulletSoftBody/btDeformableContactConstraint.cpp]: This is actually a bug, not a serious one though. I'll fix this in the next PR.

File: src/BulletSoftBody/btSoftBody.cpp (hunk 2 above)
Comment (xhan0619) [reply src/BulletSoftBody/btSoftBody.cpp]: Yes, eventually we do. But I'm still trying to understand why a smaller margin for face than for nodes helps with reducing stickiness.

**Comment by xhan0619** [comment]
> CACHE_PREV_COLLISION

Yes, we can change them to run-time flags.

**Pull request merged**